editor/inserter/menu.js

```diff
@@ -307,16 +300,54 @@ export class InserterMenu extends Component {
 		);
 	}
 
+	renderBlocks( blocks, separatorSlug ) {
+		const { instanceId } = this.props;
+		const labelledBy = separatorSlug === undefined ? null : `editor-inserter__separator-${ separatorSlug }-${ instanceId }`;
+		return (
+			<div
+				className="editor-inserter__category-blocks"
+				tabIndex="0"
+				aria-labelledby={ labelledBy }
+			>
+				{ blocks.map( ( block ) => this.getBlockItem( block ) ) }
+			</div>
+		);
+	}
+
+	renderCategory( category, blocks ) {
+		const { instanceId } = this.props;
+		return blocks && (
+			<div key={ category.slug }>
+				<div
+					className="editor-inserter__separator"
+					id={ `editor-inserter__separator-${ category.slug }-${ instanceId }` }
+					aria-hidden="true"
+				>
+					{ category.title }
+				</div>
+				{ this.renderBlocks( blocks, category.slug ) }
+			</div>
+		);
+	}
+
+	renderCategories( visibleBlocksByCategory ) {
+		return getCategories().map(
```

Comment:
I don't see how this works... visibleBlocksByCategory doesn't contain all the category information. The data returned by getCategories looks like this:

`{slug: "common", title: "Common Blocks"}`

and visibleBlocksByCategory is just an object with blocks arranged by category slug. We still need the category objects to get the title.

Also, we need to return array of react elements, and mapValues returns an object, so if we did that we'd need to iterate over the keys to render.